### ISSUE TITLE
netlink 'clash' invalid length

### ISSUE THREAD

**Issue by Yogxx** [zerolabnet/SSClash#3]
does ssclash not support tun mode?
![Screenshot_2024-12-10-13-33-19-982_com android chrome](https://github.com/user-attachments/assets/a2cf66b8-3889-4453-9676-98a95dcaa720)

**Comment by fildunsky** [comment]
> does ssclash not support tun mode?
> ![Screenshot_2024-12-10-13-33-19-982_com android chrome](https://github.com/user-attachments/assets/a2cf66b8-3889-4453-9676-98a95dcaa720)
> 

Of course, bro. SsClash itself is just a simplified luci interface for Clash (Mihomo) kernel. So whatever the kernel supports it will work. You have to set it in the config properly as per official Clash/mihomo MetaCubeX wiki.

**Comment by Yogxx** [comment]
still errors
![Screenshot_2024-12-10-15-52-42-240_com android chrome](https://github.com/user-attachments/assets/f2bdeb73-1e95-49b0-845f-624ac359b4ff)
![Screenshot_2024-12-10-15-52-30-070_com android chrome](https://github.com/user-attachments/assets/c19d0084-3d04-4c2f-85ff-3ca55f86b8f3)
![Screenshot_2024-12-10-15-52-19-440_com android chrome](https://github.com/user-attachments/assets/f574a1dc-bbbd-4b0c-8d93-f0f5cbe0e597)

**Comment by fildunsky** [comment]
Have you `opkg install kmod-tun` installed kmod-tun?

Also in the SSClash Meta tun interface second tab `Advanced Settings: Use default gateway` switch it off

Also, you have to add a new Firewall Zone
Accept all the traffic, enable masquerading, MSS clamping, allow forward from LAN zone

**Comment by Yogxx** [comment]
![Screenshot_2025-01-08-10-50-25-093_com android chrome-picsay](https://github.com/user-attachments/assets/f1598096-e295-4eb9-8397-072c18ab42a0)


It's been successful bro, after updating the core and following your instructions, and like the picture
